ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -476,6 +471,93 @@ public void replicaThreadTest() throws Exception {
           replicasToReplicate.get(remoteHost.dataNodeId).get(i).getToken());
     }
 
+    /*
+        STORE KEY CONVERTER MAPPING
+        Key     Value
+        B0      B0'
+        B1      B1'
+        B2      null
+
+        BEFORE
+        Local   Remote
+        B0'     B0
+                B1
+                B2
+
+        AFTER
+        Local   Remote
+        B0'     B0
+        B1'     B1
+                B2
+        B0 is B0' for local,
+        B1 is B1' for local,
+        B2 is null for local,
+        so it already has B0/B0'
+        B1 is transferred to B1'
+        and B2 is invalid for L
+        so it does not count as missing
+        Missing Keys: 1
+
+     */
+    Map<PartitionId, List<BlobId>> partitionIdToDeleteBlobId = new HashMap<>();
+    Map<StoreKey, StoreKey> conversionMap = new HashMap<>();
+    for (int i = 0; i < partitionIds.size(); i++) {
+      PartitionId partitionId = partitionIds.get(i);
+      List<BlobId> deleteBlobIds = new ArrayList<>();
+      partitionIdToDeleteBlobId.put(partitionId, deleteBlobIds);
+      BlobId b0 = generateRandomBlobId(partitionId);
+      deleteBlobIds.add(b0);
+      BlobId b0p = generateRandomBlobId(partitionId);
+      BlobId b1 = generateRandomBlobId(partitionId);
+      BlobId b1p = generateRandomBlobId(partitionId);
+      BlobId b2 = generateRandomBlobId(partitionId);
+      deleteBlobIds.add(b2);
+      conversionMap.put(b0, b0p);
+      conversionMap.put(b1, b1p);
+      conversionMap.put(b2, null);
+      addPutMessagesToReplicasOfPartition(Arrays.asList(b0p), Arrays.asList(localHost));
+      addPutMessagesToReplicasOfPartition(Arrays.asList(b0, b1, b2), Arrays.asList(remoteHost));
+    }
+    storeKeyConverter.setConversionMap(conversionMap);
+
+    expectedIndex =
+        assertMissingKeysAndFixMissingStoreKeys(expectedIndex, expectedIndex + 1, batchSize - 1, batchSize, 1,
```

Comment:
Does this check for exactly what keys are missing and what is not (as described in the comments)? Simply checking for the counts does not seem enough.

Reply:
The comments describe asserting the number of missing keys between local and remote replicas which is what it does.  I adapted this from the code that was already there which checked for the number of missing keys and expected token indexes.  There is another check that checks the content of the remote replica vs. the local replica and counts the different messages that are there at the end of the test.

I don't think it does a state check, like comparing the partition blob store to an expected partition blob store post replication, but I don't think the test ever checked for that.